### ISSUE TITLE
Lab 3 Test Case 3

### ISSUE THREAD

**Issue by yi-teng** [nus-cs2030/2223-s2#562]
I have one customer that leaves instead of waiting in my queue....

Any tips on how to get the customer to join the queue? I assumed my classes would not have problems since it passes the first 2 test cases.

## Screenshots

<img width="197" alt="image" src="https://user-images.githubusercontent.com/122248743/227271180-fadda1b3-6a7c-4c85-884f-112e40357379.png">

**Comment by chianxintong** [comment]
Hmm it might be helpful for you to use system.out.println to check whether your queue has been updated (check the queue size in each event). it seems like your queue is already full by the time customer 6 reaches but that shouldnt be the case (??)

**Comment by Lofthesaver** [comment]
Hi, since you are returning a `leave `rather than a `wait `from your `ArriveEvent`, it probably means that the server list isn't being updated correctly or you're not checking through the queue correctly (since this problem is when queue size > 1)

**Comment by leonardwx** [comment]
hi, it seems like server2 queue list is full which resulted customer6 to leave instead of waiting. perhaps you might want to check the wait conditions at the arrive event?

U can create an attribute like ImList of customer inside every server to keep track of the queue using size(), afterwhich for loop the ImList of Servers to search each server queue size against the qmax. if the size < qmax, then you can provide a enqueue method to add the customer inside the particular server Imlist. Then execute a Wait event for it. Inside your serve event, you have to properly dequeue the ImList of customer inside the particular server. Hope it helps :)

**Comment by shannon-jx** [comment]
hi! to add on to the previous comments (assuming you have a method minusQ() and addQ()), you can double check if you are using them correctly - minusQ() when someone leaves, but that means increase in the number of available spots in the queue AND addQ() when someone joins, but that means decrease in the number of available spots in the queue

**Comment by littlemidget3** [comment]
yeah ensure that you are removing from the queue each time your server serves, eg -1 from qlength and removing index 0 from your queue. Also ensure you are updating your list of servers after the each event. 

after don't assume the classes are working because the first 2 cases worked, some test cases dont include edge cases

**Comment by pylfmty** [comment]
can try to remove the customer in the ImList of queue of the server in serve 

server.getQueue().remove(0)
